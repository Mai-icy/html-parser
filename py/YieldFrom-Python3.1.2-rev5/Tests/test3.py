#
#   Test raising exception in initial next() call
#

def g1():
	try:
		print("Starting g1")
		yield from g2()
	finally:
		print("Finishing g1")

def g2():
	try:
		print("Starting g2")
		raise ValueError("spanish inquisition occurred")
	finally:
		print("Finishing g2")

for x in g1():
	print("Yielded", x)
