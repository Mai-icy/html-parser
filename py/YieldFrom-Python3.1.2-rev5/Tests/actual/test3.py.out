Starting g1
Starting g2
Finishing g2
Finishing g1
Traceback (most recent call last):
  File "test3.py", line 19, in <module>
    for x in g1():
  File "test3.py", line 8, in g1
    yield from g2()
  File "test3.py", line 15, in g2
    raise ValueError("spanish inquisition occurred")
ValueError: spanish inquisition occurred
